frontend/src/components/main/Main.js

```diff
@@ -1,51 +1,57 @@
 import Loader from '../loader/Loader';
-import {TokenBalance,Balance,Token,Decimals,Allowance,Approve,fetchActionsToken} from '../eth/TokenBalance';
+import {TokenBalance,Balance,Token,Decimals,Allowance,Approve,fetchActionsToken, tokenSlice} from '../eth/TokenBalance';
 import EthBalance from '../eth/EthBalance';
-import {TotalSupply,GetVault,GetStrategy, Deposit,BalanceOf,Withdraw} from '../eth/vault';
+import {TotalSupply,GetVault,GetStrategy, Deposit,Withdraw} from '../eth/vault';
 import { useState, useEffect } from 'react'
 import {ContractAddress} from '../../helpers/connector';
 import { useSelector, useDispatch } from 'react-redux';
-import {vaultSlice,fetchActions} from '../eth/vault';
+import {vaultSlice,fetchActionsVault} from '../eth/vault';
 import './Main.scss';
 import { useWeb3React } from '@web3-react/core'
 import {decimalFormat} from '../eth/helpers';
 
 const DEFAULT_BUTTON_TEXT = 'Approve';
 const ENTER_KEY_CODE = 'Enter';
 
+
 export default function Main(props) {
   const {account, library, chainId} = useWeb3React();
 
   const vaultStore = useSelector((state) => state.vault);
+  const tokenStore = useSelector((state) => state.token)
   const dispatch = useDispatch();
 
   const isButtonDisabled = props.fetching;
   const vaultContractAddress = ContractAddress("vault")
   const vault = GetVault(vaultContractAddress)
   const vaultDecimals = Decimals(vault)
-  const balanceUser = BalanceOf(vault,vaultDecimals)
+  const eth = Token(ContractAddress("eth"))
+  const dai = Token(ContractAddress("dai"))
 
   useEffect(() => {
      if(!vault){
         return
      }
      console.log("selected contract: ", vault.address)
+
      dispatch(fetchActionsToken.decimals(vault)).then(r => dispatch(vaultSlice.actions.decimals(r.payload)))
+     dispatch(fetchActionsVault.totalSupply(vault));
+     dispatch(fetchActionsVault.totalAmounts(vault));
+     dispatch(fetchActionsVault.balanceOf({account, vault}));
+     
+     dispatch(fetchActionsToken.balance({account,contract: eth})).then(r => dispatch(tokenSlice.actions.balanceDai(r.payload)));
+     dispatch(fetchActionsToken.balance({account,contract: dai})).then(r => dispatch(tokenSlice.actions.balanceDai(r.payload)));
+     dispatch(fetchActionsToken.allowanceEth({vault, account, contract: eth}));
+     dispatch(fetchActionsToken.allowanceDai({vault, account, contract: dai}));
 
-     dispatch(fetchActions.totalSupply(vault));
-     dispatch(fetchActions.totalAmounts(vault));
   }, [vault]);
 
-  const eth = Token(ContractAddress("eth"))
+
   const ethDecimals = Decimals(eth)
   const ethBalance = Balance(eth)
-  const ethAllowance = Allowance(eth, vault)
   
-  const dai = Token(ContractAddress("dai"))
   const daiDecimals = Decimals(dai)
-  const daiBalance = Balance(dai)
-  const daiAllowance = Allowance(dai, vault)
-  
+  const daiBalance = Balance(dai)  
```

Comment:
you have this in the state now, use it :)

Reply:
i'm using it,, that variable need to be deleted